File: Data Structures & Algorithms/majority-element/submission-0.java
class Solution {
    public int majorityElement(int[] nums) {
        int maj = 0;
        int cnt = 0;
        
        for (int num : nums) {
            if (cnt == 0) {
                maj = num;
            }

            if (num == maj) {
                cnt++;
            } else {
                cnt--;
            }
        }

        return maj;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        var map = new Dictionary<int, int>();
        foreach (var num in nums) {
            map[num] = map.GetValueOrDefault(num, 0) + 1;
        }

        var minHeap = new PriorityQueue<int, int>();
        foreach (var key in map.Keys) {
            minHeap.Enqueue(key, map[key]);

            if (minHeap.Count > k) {
                minHeap.Dequeue();
            }
        }

        var res = new int[k];
        for (int i = 0; i < k; i++) {
            res[i] = minHeap.Dequeue();
        }

        return res;
    }
}
